packages/siopv2-oid4vp-rp-rest-client/src/agent/SiopV2OID4VpRpRestClient.ts

```
@@ -0,0 +1,99 @@
+      this.definitionId = definitionId
+    }
+  }
+
+  private async siopClientRemoveAuthRequestSession(args: ISiopClientRemoveAuthRequestSessionArgs, context: IRequiredContext): Promise<void> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    fetch(this.uriWithBase(`/webapp/definitions/${definitionId}/auth-requests/${args.correlationId}`, baseUrl), {
+      method: 'DELETE',
+    })
+  }
+
+  private async siopClientGetAuthStatus(args: ISiopClientGetAuthStatusArgs, context: IRequiredContext): Promise<AuthStatusResponse> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const url = this.uriWithBase('/webapp/auth-status', baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    const statusResponse = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        correlationId: args.correlationId,
+        definitionId,
+      }),
+    })
+    debug(`auth status response: ${statusResponse}`)
+    try {
+      return await statusResponse.json()
+    } catch (err) {
+      throw Error(`Status has returned ${statusResponse.status}`)
+    }
+  }
+
+  private async siopClientGenerateAuthRequest(
+    args: ISiopClientGenerateAuthRequestArgs,
+    context: IRequiredContext
+  ): Promise<GenerateAuthRequestURIResponse> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    const url = this.uriWithBase(`/webapp/definitions/${definitionId}/auth-request-uri`, baseUrl)
+    const origResponse = await fetch(url)
+    const success = origResponse && origResponse.status >= 200 && origResponse.status < 400
+    if (success) {
+      return await origResponse.json()
+    }
+    throw Error(`calling ${url} returned ${origResponse.status}`)
```

Do not swallow errors at this level. This is a REST client. Whether someone using this client wants to show the actual error or create their own simpler error is up to them.